test/Main.purs

```diff
@@ -57,3 +58,27 @@ main = do
     assert $ String.words "Thor, Mímir, Ēostre, & Jörð" == [ "Thor", "Mímir", "Ēostre", "Jörð" ]
     assert $ String.words "🙃, Mímir, Ēostre, & Jörð" == [ "🙃", "Mímir", "Ēostre", "Jörð" ]
     assert $ String.words "thorMímir--Ēostre_Jörð" == [ "thor", "Mímir", "Ēostre", "Jörð" ]
+
+  -- Verified with: https://planetcalc.com/1721/
+  log "levenshtein" *> do
+    assert $ String.levenshtein "CONSERVATIONALISTS" "CONVERSATIONALISTS" == 2
+    assert $ String.levenshtein "WHIRLED" "WORLD" == 3
+    assert $ String.levenshtein "COMPLEMENT" "COMPLIMENT" == 1
+    assert $ String.levenshtein "BAZAAR" "BIZARRE" == 3
+    assert $ String.levenshtein "ACCESSARY" "ACCESSORY" == 1
+    assert $ String.levenshtein "ALGORITHMS ARE FUN" "LOGARITHMS ARE NOT" == 6
+    assert $ String.levenshtein "ASSISTANCE" "ASSISTANTS" == 2
+    assert $ String.levenshtein "ALL TOGETHER" "ALTOGETHER" == 2
+    assert $ String.levenshtein "IDENTICAL STRINGS" "IDENTICAL STRINGS" == 0
+
+  -- Verified with: http://www.algomation.com/algorithm/sorensen-dice-string-similarity
+  log "sorensen_dice_coefficient" *> do
+    assert $ String.sorensen_dice_coefficient "CONSERVATIONALISTS" "CONVERSATIONALISTS" ≅ 0.7647059
+    assert $ String.sorensen_dice_coefficient "WHIRLED" "WORLD" ≅ 0.2000000
+    assert $ String.sorensen_dice_coefficient "COMPLEMENT" "COMPLIMENT" ≅ 0.7777778
+    assert $ String.sorensen_dice_coefficient "BAZAAR" "BIZARRE" ≅ 0.36363637
+    assert $ String.sorensen_dice_coefficient "ACCESSARY" "ACCESSORY" ≅ 0.7500000
+    assert $ String.sorensen_dice_coefficient "ALGORITHMS ARE FUN" "LOGARITHMS ARE NOT" ≅ 0.5882353
+    assert $ String.sorensen_dice_coefficient "ASSISTANCE" "ASSISTANTS" ≅ 0.7777778
+    assert $ String.sorensen_dice_coefficient "ALL TOGETHER" "ALTOGETHER" ≅ 0.8000000
+    assert $ String.sorensen_dice_coefficient "IDENTICAL STRINGS" "IDENTICAL STRINGS" ≅ 1.0000000
```

Comment:
To remove the dependency on `Data.Number.Approximate`, do you think we could do something like this?

```purs
normalize :: Number -> Int
normalize n = Int.round (n * 100000.0)

normalizedCoefficient :: String -> String -> Int
normalizedCoefficient a b = normalize (String.sorensenDiceCoefficient a b)

log "sorensenDiceCoefficient" *> do
  assert $ (normalizedCoefficient "CONSERVATIONALISTS" "CONVERSATIONALISTS") == (normalize 0.7647059)
```